cmd/dockerd/daemon.go

```diff
@@ -455,6 +455,12 @@ func loadDaemonCliConfig(opts *daemonOptions) (*config.Config, error) {
 		conf.TLSVerify = conf.TLS
 	}
 
+	if opts.Validate {
+		// If config wasn't OK we wouldn't have made it this far.
+		logrus.Infof("Config OK")
+		os.Exit(0)
```

Comment:
IMO, we should be careful use `os.Exit` and that the program exit immediately in big program, this may cause some `defer` doesn't perform and then left with some uncleaned resoure.  https://gobyexample.com/exit It's better to return a state  to the caller.

Reply:
Oh, good one, actually didn't know that.

Yes, looks like that would cause some things to not be run https://github.com/moby/moby/blob/94c07441c233ed4b738b97ca19774912f728075e/cmd/dockerd/daemon.go#L77-L83

Thinking how we can achieve this; we could return a "special" error, but that may need handling of that error is various places to make sure we still still the daemon, but with a `zero` exit code.